src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -1,49 +1,97 @@
 import { ItemData, ItemDataSource } from '../../services/item-datasource.service';
 import { FormBuilder, Validators } from '@angular/forms';
 import { Subscription } from 'rxjs';
 import { filter } from 'rxjs/operators';
 import { FetchError, Fetching, isReady, Ready } from '../../../../shared/helpers/state';
+import { UpdateItemStringService } from '../../http-services/update-item-string.service';
+import { Item } from '../../http-services/get-item-by-id.service';
+import { TOAST_LENGTH } from '../../../../shared/constants/global';
+import { MessageService } from 'primeng/api';
+import { ERROR_MESSAGE } from '../../../../shared/constants/api';
 
 
 @Component({
   selector: 'alg-item-edit',
   templateUrl: './item-edit.component.html',
-  styleUrls: ['./item-edit.component.scss']
+  styleUrls: ['./item-edit.component.scss'],
+  providers: [MessageService]
 })
 export class ItemEditComponent implements OnDestroy {
+  item: Item;
   itemForm = this.formBuilder.group({
     // eslint-disable-next-line @typescript-eslint/unbound-method
     title: ['', [Validators.required, Validators.minLength(3),]],
   });
   itemLoadingState$ = this.itemDataSource.state$;
 
-  itemSubscription: Subscription;
+  subscriptions: Subscription[] = [];
 
   constructor(
     private currentContent: CurrentContentService,
     private itemDataSource: ItemDataSource,
     private formBuilder: FormBuilder,
+    private updateItemStringService: UpdateItemStringService,
+    private messageService: MessageService
   ) {
     this.currentContent.editState.next('editing');
     this.getCurrentItem();
+    this.onSubmit();
   }
 
   ngOnDestroy() {
     this.currentContent.editState.next('non-editable');
-    this.itemSubscription.unsubscribe();
+    this.subscriptions.forEach(subscription => subscription.unsubscribe());
   }
 
   getCurrentItem(): void {
-    this.itemSubscription = this.itemLoadingState$.pipe(
+    this.subscriptions.push(this.itemLoadingState$.pipe(
       filter<Ready<ItemData> | Fetching | FetchError, Ready<ItemData>>(isReady),
     ).subscribe(state => {
-      const item = state.data.item;
+      this.item = state.data.item;
       this.itemForm.patchValue({
-        title: item.string.title
+        title: this.item.string.title
       });
+    }));
+  }
+
+  onSubmit(): void {
+    this.subscriptions.push(this.currentContent.editAction$
+      .pipe(filter(action => action === EditAction.Save))
+      .subscribe(_action => this.saveInput()));
+  }
+
+  errorToast() :void {
+    this.messageService.add({
+      severity: 'error',
+      summary: 'Error',
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
     });
   }
 
+  successToast() : void {
+    this.messageService.add({
+      severity: 'success',
+      summary: 'Success',
+      detail: 'Changes successfully saved.',
+      life: TOAST_LENGTH,
+    });
+  }
+
+  saveInput(): void {
+    this.updateItemStringService.updateItem(
+      this.item.id,
+      this.itemForm.value
```

Comment:
It might work at first (except if you add the id as I suggest above) but probably you'll need to extract properly the value (I think, that might be done later maybe).

Reply:
If all the inputs are named correctly it should work as itemForm.value return an object that match what http.put needs. But I could be missing something.

Reply:
Yes, that should be ok for now.